### ISSUE TITLE
get_tree().quit() not doing anything in web exports

### ISSUE THREAD

**Issue by Zireael07** [godotengine/godot#23010]
**Godot version:**
3.0.3 (but I have seen this also with Tanks of Freedom, which is a 2.1.x-based project)

**OS/device including version:**
Win 7 x64

**Issue description:**
If you have a game exported to web that uses get_tree().quit to, well, quit, it will do nothing in a web export. Functions that should run on QUIT_NOTIFICATIONS will not run and the game will stay in browser, still active...

**Steps to reproduce:**
https://tof.p1x.in/html5/

wait for game to load, click quit game (top right of screen)

**Minimal reproduction project:**
Do you really want one for such a reproducible issue as this?

**Comment by Zappo-II** [comment]
By the way, after call to get_tree().quit collision detection is not working any longer...but that's irrelevant since the whole game should not work any longer in the first place...

**Comment by bbooth87** [comment]
I'm using 3.1.1 and having the same issue. Is there a resolution or workaround?

**Comment by sparkart** [comment]
Try calling the the `window.close()` function in Javascript.

Godot has a Javascript singleton to execute Javascript:
https://docs.godotengine.org/en/3.1/classes/class_javascript.html

**Comment by ndarilek** [comment]
Here's what I tried:

```
func _on_Quit_pressed():
  if OS.has_feature('JavaScript'):
    JavaScript.eval("window.close()")
  get_tree().quit()
```

That throws the following in Firefox's dev console:

`Scripts may not close windows that were not opened by script.`

Has anyone found a workaround?

**Comment by Calinou** [comment]
@ndarilek According to [this Stack Overflow question](https://stackoverflow.com/questions/2076299/how-to-close-current-tab-in-a-browser-window), this can be worked around by using a `<button>` element (which needs to be part of the DOM first). This is probably difficult to integrate in a WebAssembly-based engine.

**Comment by ndarilek** [comment]
Got it. I'm more interested in allowing the game to be stopped, than in 
closing the window. Wasn't sure if there was a lower-level function 
other than `get_tree().quit()` I might try.


Thanks.

**Comment by ndarilek** [comment]
Could it have to do with this in platform/javascript/detect.py?:

```
    # TODO: Reevaluate usage of this setting now that engine.js manages engine runtime.
    env.Append(LINKFLAGS=['-s', 'NO_EXIT_RUNTIME=1'])
```

From https://emscripten.org/docs/getting_started/FAQ.html:

"By default Emscripten sets EXIT_RUNTIME=0, which means that we don’t include code to shut down the runtime. That means that when main() exits, we don’t flush the stdio streams, or call the destructors of global C++ objects, or call atexit callbacks. This lets us emit smaller code by default, and is normally what you want on the web: even though main() exited, you may have something asynchronous happening later that you want to execute.

In some cases, though, you may want a more “commandline” experience, where we do shut down the runtime when main() exits. You can build with -s EXIT_RUNTIME=1, and then we will call atexits and so forth."

What would re-evaluating usage of this setting look like? Just changing it and seeing if exiting works? Or is there more to it?

**Comment by akien-mga** [comment]
Fixed in `master` by #38587 (and in `3.2` by #39604).
